saleor/graphql/checkout/mutations.py

```diff
@@ -1029,6 +1045,175 @@ def perform_mutation(
         return CheckoutShippingMethodUpdate(checkout=checkout)
 
 
+class CheckoutDeliveryMethodUpdate(BaseMutation):
+    checkout = graphene.Field(Checkout, description="An updated checkout.")
+
+    class Arguments:
+        token = UUID(description="Checkout token.", required=False)
+        delivery_method_id = graphene.ID(
+            description="Delivery Method ID (`Warehouse` ID or `ShippingMethod` ID).",
+            required=False,
+        )
+
+    class Meta:
+        description = (
+            "Updates the delivery method (shipping method or pick up point)"
+            "of the checkout."
+        )
+        error_type_class = CheckoutError
+        error_type_field = "checkout_error"
```

Comment:
We could `error_type_field` remove it. This field is responsible for adding deprecated errors to the mutaion. We could add new mutations without old errors.

Reply:
Done.